nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/marshallers/json/enrichers/AbstractJsonEnricher.java

```diff
@@ -51,8 +55,18 @@ public boolean accept(Class<?> clazz, Type genericType, MediaType mediatype) {
     }
 
     @Override
-    public void write(Enriched<EntityType> enrichable, JsonGenerator jg) throws IOException {
-        write(jg, enrichable.getEntity());
+    public void write(Enriched<EntityType> enrichable, JsonGenerator jg) {
+        try {
+            write(jg, enrichable.getEntity());
+        } catch (Exception e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException("interrupted", e); // NOSONAR
```

Comment:
You could use this:
```java
org.nuxeo.common.utils.ExceptionUtils.checkInterrupt(e);
```
and get rid of these three lines + if / else.

Reply:
Actually I prefer that we don't call checkInterrupt anymore, as it confuses tools that look for specific InterruptedException management.